source/beamline/BioXAS/BioXASMasterValves.cpp

```diff
@@ -0,0 +1,109 @@
+#include "BioXASMasterValves.h"
+#include "beamline/BioXAS/BioXASFrontEndValves.h"
+#include "beamline/BioXAS/BioXASSideValves.h"
+#include "beamline/BioXAS/BioXASMainValves.h"
+#include "beamline/BioXAS/BioXASImagingValves.h"
+
+BioXASMasterValves::BioXASMasterValves(QObject *parent) :
+	BioXASValves("BioXASMasterValves", parent)
+{
+	// Initialize class variables.
+
+	frontEndValves_ = 0;
+	sideValves_ = 0;
+	mainValves_ = 0;
+	imagingValves_ = 0;
+}
+
+BioXASMasterValves::~BioXASMasterValves()
+{
+
+}
+
+bool BioXASMasterValves::isOpen() const
+{
+	bool open = ( isConnected() && BioXASValves::isOpen() );
```

Comment:
Yes, you're right. Or, er, it should. Looking at BioXASValves, I don't actually see a connectivity check. I'll fix this, and check the other descendants of BioXASBiStateGroup for the same error.